src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+            with open(self.__CACHE_FILE, 'wb') as file:
+                pickle.dump(result, file)
+
+        return result
+
+    def load(self) -> List[AbstractTransaction]:
+        (trade_history, ledger) = self._gather_api_data()
+
+        result = {'in': [], 'out': [], 'intra': []}
+
+        unhandled_types = dict()
+        for key, value in ledger.items():
+            if value[_TYPE] == _WITHDRAWAL or value[_TYPE] == _DEPOSIT:
+                result[_INTRA].append(key)
+            elif value[_TYPE] == _TRADE and 'USD' not in value[_ASSET]:
+                if float(value[_AMOUNT]) > 0: result[_IN].append(key)
+                else: result[_OUT].append(key)
+            elif value[_TYPE] == _MARGIN:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _ROLLOVER:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _TRANSFER:
+                result[_IN].append(key)
+            elif value[_TYPE] == _SETTLED:
+                # ignorable in terms of in/out/intra
+                pass
+            elif value[_TYPE] == _CREDIT:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _STAKING:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _SALE:
+                # 'credit not implemented'
+                pass
+            else:
+                unhandled_types.update({value[_TYPE]: key})
+        self.__logger.debug(f"unhandled types of the ledger={unhandled_types}")
+
+        return self._compute_tx_set(trade_history, ledger, result)
+
+    def _compute_tx_set(self, trade_history, ledger, processed_transactions) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        transactions = processed_transactions[_IN] + processed_transactions[_OUT] + processed_transactions[_INTRA]
+        for key in transactions:
+            record = ledger[key]
+            timestamp_value: datetime = datetime.fromtimestamp(float(record[_TIMESTAMP]))
+            timestamp_value = timestamp_value.astimezone(self.__timezone)
+            timestamp_value = self.__timezone.normalize(timestamp_value)
+
+            is_fiat_asset = record[_ASSET] in _FIAT_SET or 'USD' in record[_ASSET]
+
+            amount = abs(float(record[_AMOUNT]))
+            asset = self.baseId_to_base[record[_ASSET]]
+            kwargs = {
+                'unique_id': Keyword.UNKNOWN.value,
+                'plugin': self.__PLUGIN_NAME,
+                'raw_data': str(record),
+                'timestamp': str(timestamp_value),
+                'spot_price': '0',
+                'asset': asset,
+                'notes': key,
+            }
+
+            if record[_TYPE] == _WITHDRAWAL or record[_TYPE] == _DEPOSIT:
+                is_deposit = record[_TYPE] == _DEPOSIT
+                is_withdrawal = record[_TYPE] == _WITHDRAWAL
+
+                # Intra
+                kwargs.update(
```

Comment:
I would advise against reusing the dictionary because it can lead to subtle and hard-to-fix bugs: it's not worth the space savings in the code in my opinion. Just spell out all the arguments directly in the call to InTransaction, OutTransaction and IntraTransaction.